server/app/auth/ProfileFactory.java

```diff
@@ -50,6 +57,13 @@ public CiviFormProfile wrapProfileData(CiviFormProfileData p) {
     return new CiviFormProfile(dbContext, httpContext, p);
   }
 
+  public Optional<ApiKey> retrieveApiKey(String keyId) {
```

Comment:
Re the slack discussion, might be nice to add a comment in this file for why this makes sense, and also to dissuade adding profile unrelated code here (which this may appear to be)

Reply:
doc'd the method